eq-author/src/components/modals/ConfirmationModal/index.js

```diff
@@ -73,16 +77,42 @@ const ConfirmationModal = ({
         <Button variant="secondary" onClick={handleCancel}>
           {cancelText}
         </Button>
-        <Button variant="primary" onClick={handleConfirm}>
+        <Button variant="primary" onClick={handleConfirm} autoFocus>
           {confirmText}
         </Button>
       </ButtonGroupStyled>
     </Modal>
   );
 };
 
+export const useConfirmationModal = ({ action, ...props }) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleConfirm = useCallback(() => {
+    action();
+    setIsOpen(false);
+  }, [action]);
+
+  const handleCancel = useCallback(() => setIsOpen(false));
```

Comment:
```suggestion
  const handleCancel = useCallback(() => setIsOpen(false), []);
```
It never needs to change because it's using a static value or you want it to change everytime?